core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java

```diff
@@ -120,15 +118,20 @@ private FileIO io(String location) {
       try {
         io = CatalogUtil.loadFileIO(impl, properties, conf);
       } catch (IllegalArgumentException e) {
-        LOG.warn("Failed to load FileIO implementation: {}, falling back to {}", impl, FALLBACK_IMPL, e);
-        try {
-          // couldn't load the normal class, fall back to HadoopFileIO
-          io = CatalogUtil.loadFileIO(FALLBACK_IMPL, properties, conf);
-        } catch (IllegalArgumentException suppressed) {
-          LOG.warn("Failed to load FileIO implementation: {} (fallback)", FALLBACK_IMPL, suppressed);
-          // both attempts failed, throw the original exception with the later exception suppressed
-          e.addSuppressed(suppressed);
+        if (impl.equals(FALLBACK_IMPL)) {
+          // no implementation to fall back to, throw the exception
           throw e;
+        } else {
+          // couldn't load the normal class, fall back to HadoopFileIO
+          LOG.warn("Failed to load FileIO implementation: {}, falling back to {}", impl, FALLBACK_IMPL, e);
+          try {
+            io = CatalogUtil.loadFileIO(FALLBACK_IMPL, properties, conf);
+          } catch (IllegalArgumentException suppressed) {
+            LOG.warn("Failed to load FileIO implementation: {} (fallback)", FALLBACK_IMPL, suppressed);
+            // both attempts failed, throw the original exception with the later exception suppressed
+            e.addSuppressed(suppressed);
+            throw e;
+          }
```

Comment:
Does catching and performing work in here generate any warnings?

I _think_ that's only for returning, but checking.

Reply:
I'm not quite sure what you mean